examples/resnet/src/train.jl

```diff
@@ -0,0 +1,20 @@
+using Flux, FluxMPI, CUDA, Metalhead
+
+FluxMPI.Init()
+
+model = ResNet50() |> gpu
+ps = Flux.params(model)
+broadcast_parameters(model; root_rank = 0)
+
+batchsize = 8
+data = [(rand(Float32, 224, 224, 3, batchsize) |> gpu, onehotbatch(rand(1:1000), 1:1000)) |> gpu for _ in 1:3]
+
+opt = DistributedOptimiser(ADAM())
+
+loss(x, y, m) = Flux.Losses.logitcrossentropy(m(x), y)
+
+for (i, (x, y)) in enumerate(data)
+    @info "Starting batch $i ..."
```

Comment:
using `@info` in MPI code would not play well. You need to use `clean_println` / `clean_print` or put a check for `local_rank() == 0`